### ISSUE TITLE
importing tmap in python causes core dump, x86-64 version on both mac and linux

### ISSUE THREAD

**Issue by drmatthewclark** [reymond-group/tmap#66]
behaivior:  - just import tmap to reproduce.  using python3.6 for compatibility; it used to work.

(tmap) mclark@metzora:~$ python
Python 3.6.13 |Anaconda, Inc.| (default, Jun  4 2021, 14:25:59) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tmap
Illegal instruction (core dumped)

OS

Linux metzora 6.11.4-101.fc39.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Oct 20 15:02:40 UTC 2024 x86_64 GNU/Linux

environment:
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
ca-certificates           2024.9.24            h06a4308_0  
certifi                   2021.5.30        py36h06a4308_0  
ld_impl_linux-64          2.40                 h12ee557_0  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 11.2.0               h1234567_1  
libgomp                   11.2.0               h1234567_1  
libstdcxx-ng              11.2.0               h1234567_1  
ncurses                   6.4                  h6a678d5_0  
ogdf                      1.2.0                h2bc3f7f_0    tmap
openssl                   1.1.1w               h7f8727e_0  
pip                       21.2.2           py36h06a4308_0  
python                    3.6.13               h12debd9_1  
readline                  8.2                  h5eee18b_0  
setuptools                58.0.4           py36h06a4308_0  
sqlite                    3.45.3               h5eee18b_0  
tk                        8.6.14               h39e8969_0  
tmap                      1.0.6            py36h2bc3f7f_0    tmap
wheel                     0.37.1             pyhd3eb1b0_0  
xz                        5.4.6                h5eee18b_1  
zlib                      1.2.13               h5eee18b_1  

stack trace - 
                                                 Module /home/mclark/.conda/envs/tmap/lib/python3.6/lib-dynload/readline.cpython-36m-x86_64-linux-gnu.so without build-id.
                                                 Module /home/mclark/.conda/envs/tmap/lib/python3.6/lib-dynload/readline.cpython-36m-x86_64-linux-gnu.so
                                                 Stack trace of thread 18585:
                                                 #0  0x00007fbf6002339c n/a (/home/mclark/.conda/envs/tmap/lib/python3.6/site-packages/tmap.cpython-36m-x86_64-linux-gnu.so + 0x22339c)
                                                 ELF object binary architecture: AMD x86-64
Oct 24 16:45:08 metzora (sd-parse-elf)[18588]: Could not parse number of program headers from core file: invalid `Elf' handle
Oct 24 16:45:08 metzora (sd-parse-elf)[18588]: Could not parse number of program headers from core file: invalid `Elf' handle
Oct 24 16:45:08 metzora (sd-parse-elf)[18588]: Could not parse number of program headers from core file: invalid `Elf' handle
Oct 24 16:45:08 metzora (sd-parse-elf)[18588]: Could not parse number of program headers from core file: invalid `Elf' handle
Oct 24 16:45:08 metzora (sd-parse-elf)[18588]: Could not parse number of program headers from core file: invalid `Elf' handle

on macos using the x86_64 ithas the same issue
Python 3.6.13 |Anaconda, Inc.| (default, Feb 23 2021, 12:58:59) 
[GCC Clang 10.0.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> 
>>> import tmap
zsh: illegal hardware instruction  python

**Comment by doublethefish** [comment]
Try pinning your tmap-viz version to 1.0.15. That is worth a try as you're using conda, and 1.0.15 is the last version built via conda IIRC.

You are probably aware that Python 3.6 has been end-of-lifed, and isn't officially  supported by the Python Software Foundation, or conda. I know it's sometimes hard to update your toolchain, but you should consider it.
see https://devguide.python.org/versions/

 But if, I get a second I will try it out as we haven't yet gotten versions building for 3.11 or 3.12 building yet! (this is because the OSX portion of OGDF takes ~6 hours to build on github-actions and it is a massive time-sink).

**Comment by doublethefish** [comment]
I do not get your segfault, but I do get another incompatability with 3.6. `tmap-viz` uses the `dataclasses` modules which seems to have been added in Python 3.7. I assume that the previous maintainers didn't test 1.0.15 against 3.6 otherwise we wouldn't have shipped a 3.6 wheel.

On OSX with Python 3.6.15 (NOT the conda version) I get the following error:
```zsh
(.pyenv) dd /tmp/3.6$ python3 -m pip install "tmap-viz==1.0.15"
Collecting tmap-viz==1.0.15
  Downloading tmap_viz-1.0.15-cp36-cp36m-macosx_10_9_x86_64.whl (4.7 MB)
     |████████████████████████████████| 4.7 MB 2.2 MB/s
Installing collected packages: tmap-viz
  Attempting uninstall: tmap-viz
    Found existing installation: tmap-viz 1.0.16
    Uninstalling tmap-viz-1.0.16:
      Successfully uninstalled tmap-viz-1.0.16
Successfully installed tmap-viz-1.0.15
(.pyenv) dd /tmp/3.6$ python3 -c "import tmap"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/private/tmp/3.6/.pyenv/lib/python3.6/site-packages/tmap/__init__.py", line 3, in <module>
    from .embedding import embed
  File "/private/tmp/3.6/.pyenv/lib/python3.6/site-packages/tmap/embedding.py", line 2, in <module>
    from tmap.core import TMAPEmbedding
  File "/private/tmp/3.6/.pyenv/lib/python3.6/site-packages/tmap/core/__init__.py", line 1, in <module>
    from .tmap_embedding import TMAPEmbedding
  File "/private/tmp/3.6/.pyenv/lib/python3.6/site-packages/tmap/core/tmap_embedding.py", line 2, in <module>
    from .line import Line
  File "/private/tmp/3.6/.pyenv/lib/python3.6/site-packages/tmap/core/line.py", line 1, in <module>
    from dataclasses import dataclass
ModuleNotFoundError: No module named 'dataclasses'
(.pyenv) dd /tmp/3.6$ python3 --version
Python 3.6.15
```

I am going to close this as "won't fix", but I would be more than willing to accept a pull-request, but bear in mind that we have release issues build tmap-viz on OSX.

**Comment by drmatthewclark** [comment]
I used python 3.6 because installing with conda actually installed v3.6 it for me.
